### PR TITLE
Embed common structs

### PR DESCRIPTION
**Pull request by eshulman2**
Create common structs and embed them in the options files

### CODE REVIEW

File: api/v1beta1/common.go [hunk 1]
```diff
@@ -33,3 +39,72 @@ type extraConfigmapsMounts struct {
...
+
+	// Map of hashes to track e.g. job status
+	Hash map[string]string `json:"hash,omitempty"`
+
+	// Conditions
+	Conditions condition.Conditions `json:"conditions,omitempty" optional:"true"`
+
+	// NetworkAttachments status of the deployment pods
+	NetworkAttachments map[string][]string `json:"networkAttachments,omitempty"`
+}
```
Comment (lpiwowar) [inline api/v1beta1/common.go:110]: There should be a new line.

**Review by lpiwowar — COMMENTED** [review]
The direction looks good to me! I have two suggestions about using this PR for introducing the `ExtraConfigmapsMounts`, `NodeSelector` and `Toleration` parameters into CRs without these parameters (separate commits, same PR). What do you think, @eshulman2? It would simplify the `common.go` and I feel like it would ensure that we cleaned up the parameters properly. I can imagine that those parameters might come handy in the other CRs as well.

I'm taking a look at the issue with `zz_generated.deepcopy.go`.

**Comment by lpiwowar** [comment]
Some pre-commit checks failed:
```
pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/pkg/tobiko/volumes.go b/pkg/tobiko/volumes.go
index 035162d..6af0887 100644
--- a/pkg/tobiko/volumes.go
+++ b/pkg/tobiko/volumes.go
@@ -21,7 +21,6 @@ func GetVolumes(
 	var tlsCertificateMode int32 = 0444
 	var publicInfoMode int32 = 0744
 
-
 	volumes := []corev1.Volume{
 		{
 			Name: "tobiko-config",
```

If you want to run pre-commit checks locally before you push the changes follow the steps from README.md [1].

[1] [here](https://github.com/openstack-k8s-operators/test-operator?tab=readme-ov-file#contributing)

File: api/bases/test.openstack.org_ansibletests.yaml [hunk 2]
```diff
@@ -114,20 +114,65 @@ spec:
...
                   type: object
                 type: array
+              nodeSelector:
+                additionalProperties:
+                  type: string
+                description: This value contains a nodeSelector value that is applied
+                  to test pods spawned by the test operator.
+                type: object
               openStackConfigMap:
-                default: openstack-config
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_ansibletests.yaml:118]: why are we removing the default value?

File: api/bases/test.openstack.org_ansibletests.yaml [hunk 3]
```diff
@@ -114,20 +114,65 @@ spec:
...
                 description: OpenStackConfigMap is the name of the ConfigMap containing
                   the clouds.yaml
                 type: string
               openStackConfigSecret:
-                default: openstack-config-secret
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_ansibletests.yaml:123]: why are we removing the default value?

File: api/bases/test.openstack.org_horizontests.yaml [hunk 4]
```diff
@@ -60,16 +60,38 @@ spec:
...
                 default: 0
                 description: BackoffLimimt allows to define the maximum number of
-                  retried executions.
+                  retried executions (defaults to 6).
                 format: int32
                 type: integer
               containerImage:
                 default: ""
-                description: Container image for horizontest
+                description: Container image for AnsibleTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_horizontests.yaml:68]: This does not look right this HorizonTest CRD.

File: api/bases/test.openstack.org_horizontests.yaml [hunk 5]
```diff
@@ -132,10 +201,8 @@ spec:
             - dashboardUrl
             - horizonRepoBranch
             - repoUrl
-            - storageClass
             type: object
           status:
-            description: HorizonTestStatus defines the observed state of HorizonTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_horizontests.yaml:138]: Why is this description removed?

File: api/bases/test.openstack.org_tempests.yaml [hunk 6]
```diff
@@ -71,8 +71,8 @@ spec:
                   files. Those get added to the service config dir in /etc/test_operator/<file>
                 type: object
               containerImage:
-                description: An URL of a tempest container image that should be used
-                  for the execution of tempest tests.
+                default: ""
+                description: Container image for AnsibleTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:75]: This does not look right. This is CRD for `Tempest`.

File: api/bases/test.openstack.org_tempests.yaml [hunk 7]
```diff
@@ -117,12 +117,10 @@ spec:
                   to test pods spawned by the test operator.
                 type: object
               openStackConfigMap:
-                default: openstack-config
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:120]: Why are we removing the default value here?

File: api/bases/test.openstack.org_tempests.yaml [hunk 8]
```diff
@@ -117,12 +117,10 @@ spec:
...
                 description: OpenStackConfigMap is the name of the ConfigMap containing
                   the clouds.yaml
                 type: string
               openStackConfigSecret:
-                default: openstack-config-secret
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:125]: Why are we removing the default value here?

File: api/bases/test.openstack.org_tempests.yaml [hunk 9]
```diff
@@ -134,8 +132,7 @@ spec:
                 type: boolean
               storageClass:
                 default: local-storage
-                description: Name of a storage class that is used to create PVCs for
-                  logs storage. Required if default storage class does not exist.
+                description: StorageClass used to create PVCs that store the logs
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:135]: Since we are in the mode of polishing stuff let's generalize it a little bit.
```suggestion
                description: StorageClass used to create any test-operator related PVCs.
```

File: api/bases/test.openstack.org_tempests.yaml [hunk 10]
```diff
@@ -494,9 +492,31 @@ spec:
                         dir in /etc/test_operator/<file>
                       type: object
                     containerImage:
-                      description: An URL of a tempest container image that should
-                        be used for the execution of tempest tests.
+                      default: ""
+                      description: Container image for AnsibleTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:496]: This does not look right. This is `Tempest` CR. 

Maybe something like this?
```suggestion
                      description: A URL of a container image that should be used by the test-operator for tests execution.
```

File: api/bases/test.openstack.org_tempests.yaml [hunk 11]
```diff
@@ -529,9 +549,9 @@ spec:
                       pattern: ^[a-z0-9]
                       type: string
                     storageClass:
-                      description: Name of a storage class that is used to create
-                        PVCs for logs storage. Required if default storage class does
-                        not exist.
+                      default: local-storage
+                      description: StorageClass used to create PVCs that store the
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:553]: ditto

File: api/bases/test.openstack.org_tempests.yaml [hunk 12]
```diff
@@ -836,13 +856,8 @@ spec:
                   - stepName
                   type: object
                 type: array
-            required:
-            - containerImage
-            - openStackConfigMap
-            - openStackConfigSecret
             type: object
           status:
-            description: TempestStatus defines the observed state of Tempest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tempests.yaml:845]: Why are we removing this?

File: api/bases/test.openstack.org_tobikoes.yaml [hunk 13]
```diff
@@ -56,7 +56,7 @@ spec:
                 type: string
               containerImage:
                 default: ""
-                description: Container image for tobiko
+                description: Container image for AnsibleTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tobikoes.yaml:59]: ditto

File: api/bases/test.openstack.org_tobikoes.yaml [hunk 14]
```diff
@@ -178,8 +201,31 @@ spec:
                       description: tobiko.conf
                       type: string
                     containerImage:
-                      description: Container image for tobiko
+                      default: ""
+                      description: Container image for AnsibleTest
```
Comment (lpiwowar) [inline api/bases/test.openstack.org_tobikoes.yaml:205]: ditto

File: api/v1beta1/common.go [hunk 15]
```diff
@@ -33,3 +38,66 @@ type extraConfigmapsMounts struct {
 	// Config map subpath for mounting, defaults to configmap root.
 	SubPath string `json:"subPath"`
 }
+
+type CommonOptions struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="local-storage"
+	// StorageClass used to create PVCs that store the logs
```
Comment (lpiwowar) [inline api/v1beta1/common.go:46]:
```suggestion
        // StorageClass used to create any test-operator related PVCs.
```

File: api/v1beta1/common.go [hunk 16]
```diff
@@ -33,3 +38,66 @@ type extraConfigmapsMounts struct {
...
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// This value contains a toleration that is applied to pods spawned by the
+	// test pods that are spawned by the test-operator.
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type CommonOpenstackConfig struct {
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Optional
```
Comment (lpiwowar) [inline api/v1beta1/common.go:81]: Maybe we should add the default value back?

```
// +kubebuilder:default=openstack-config
```

File: api/v1beta1/common.go [hunk 17]
```diff
@@ -33,3 +38,66 @@ type extraConfigmapsMounts struct {
...
+	// OpenStackConfigMap is the name of the ConfigMap containing the clouds.yaml
+	OpenStackConfigMap *string `json:"openStackConfigMap,omitempty"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Optional
```
Comment (lpiwowar) [inline api/v1beta1/common.go:86]: Maybe we should add the default value back?

```
// +kubebuilder:default=openstack-config-secret
```

File: api/v1beta1/tobiko_types.go [hunk 18]
```diff
@@ -18,238 +18,160 @@ package v1beta1
...
-
-        // +kubebuilder:validation:Optional
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// This value contains a nodeSelector value that is applied to test pods
-	// spawned by the test operator.
-	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	CommonOptions              `json:",inline"`
 
 	// +kubebuilder:validation:Optional
-        // +kubebuilder:default=false
```
Comment (lpiwowar) [inline api/v1beta1/tobiko_types.go:53]: I think it is good idea to fix the lint issues. Can we make it a separate commit as part of this PR (since this is not related to the `common.go` effort)?

File: controllers/tempest_controller.go [hunk 19]
```diff
@@ -332,11 +332,11 @@ func (r *TempestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 	//                 It will simplify the logic and duplicite code (Tempest vs Tobiko)
 	if externalWorkflowCounter < len(instance.Spec.Workflow) {
 		if instance.Spec.Workflow[externalWorkflowCounter].NodeSelector != nil {
```
Comment (lpiwowar) [inline controllers/tempest_controller.go:334]: Will this condition ever be true after the change? Right now each parameter of the workflow section should be a pointer to easily check whether the value was set or no. Can we maybe keep it this way for now?

**Review by lpiwowar — CHANGES_REQUESTED** [review]
So far it looks good :+1: . Thank you for fixing the linting issue. It was in my nightmares for quite some time:). I've added couple of comments. Also, I have one small request, please. Can you split up the PR into multiple commits? The PR is getting quite chunky and not all the changes are tightly connected to the changes in `common.go`. If I was writing the PR I would introduce the following commits for it (e.g., [1]):

- commit that introduces the common sections for common.go
- commit that introduces `NodeSelector` and `Toleration` parameter
- commit that introduces `ExtraConfigmapsMounts`
- commit that fixes the linting issues

[1] [here](https://github.com/openstack-k8s-operators/test-operator/pull/184/commits)

File: api/bases/test.openstack.org_ansibletests.yaml (hunk 2 above)
Comment (eshulman2) [reply api/bases/test.openstack.org_ansibletests.yaml:118]: re-added

File: api/bases/test.openstack.org_ansibletests.yaml (hunk 3 above)
Comment (eshulman2) [reply api/bases/test.openstack.org_ansibletests.yaml:123]: re-added

File: controllers/tempest_controller.go (hunk 19 above)
Comment (eshulman2) [reply controllers/tempest_controller.go:334]: yes, it can be true having that you can still add node selector to the workflow itself

Comment (lpiwowar) [reply controllers/tempest_controller.go:334]: Yes, that's true. Thanks!

I paused here because I wasn't :100:  sure whether we can compare  `maps` and `slices` to `nil` without visiting some unwanted go behaviors. But we should be good here :+1:

File: api/v1beta1/tobiko_types.go [hunk 20]
```diff
@@ -18,238 +18,160 @@ package v1beta1
...
-	// This value contains a toleration that is applied to pods spawned by the
-	// test pods that are spawned by the test-operator.
-	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// +kubebuilder:default:="py3"
+	// Test environment
+	Testenv string `json:"testenv"`
 
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-        // +kubebuilder:default:="py3"
```
Comment (lpiwowar) [inline api/v1beta1/tobiko_types.go:68]: The majority of the linting fixes are not included in the `Fix linting` commit. Can we move these linting fixes into this commit? Maybe if you perform the linting fix as first commit it might make things easier for you? [1]

[1] https://github.com/openstack-k8s-operators/test-operator/pull/200/commits/2667eb3fe0b7d157ac0c94aa54204cd219019842

File: api/v1beta1/common.go [hunk 21]
```diff
@@ -33,3 +38,69 @@ type extraConfigmapsMounts struct {
...
+	StorageClass string `json:"storageClass"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=""
+	// A URL of a container image that should be used by the test-operator for tests execution.
+	ContainerImage string `json:"containerImage"`
+
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// BackoffLimimt allows to define the maximum number of retried executions (defaults to 6).
```
Comment (lpiwowar) [inline api/v1beta1/common.go:56]: This mistake was already there but since there are changes that are required as part of this PR we can update this as well.

```suggestion
	// BackoffLimimt allows to define the maximum number of retried executions (defaults to 0).
```

File: config/crd/bases/test.openstack.org_horizontests.yaml [hunk 22]
```diff
@@ -132,10 +203,9 @@ spec:
             - dashboardUrl
             - horizonRepoBranch
             - repoUrl
-            - storageClass
             type: object
           status:
-            description: HorizonTestStatus defines the observed state of HorizonTest
+            description: CommonTestStatus defines the observed state of the controller
```
Comment (lpiwowar) [inline config/crd/bases/test.openstack.org_horizontests.yaml:208]: I know that this is a nit picking but since this is about cleaning up the CRs I'm trying to pick all the problems. `CommonTestStatus` is not a name that is facing the users of the test-operator but `status`. Try using `oc explain horizontest.status`.

```suggestion
            description: Status defines the observed state of the controller
```

File: config/crd/bases/test.openstack.org_tempests.yaml [hunk 23]
```diff
@@ -836,13 +863,9 @@ spec:
...
                   type: object
                 type: array
-            required:
-            - containerImage
-            - openStackConfigMap
-            - openStackConfigSecret
             type: object
           status:
-            description: TempestStatus defines the observed state of Tempest
+            description: CommonTestStatus defines the observed state of the controller
```
Comment (lpiwowar) [inline config/crd/bases/test.openstack.org_tempests.yaml:868]: ditto

File: config/crd/bases/test.openstack.org_tobikoes.yaml [hunk 24]
```diff
@@ -277,11 +327,9 @@ spec:
                   - stepName
                   type: object
                 type: array
-            required:
-            - storageClass
             type: object
           status:
-            description: TobikoStatus defines the observed state of Tobiko
+            description: CommonTestStatus defines the observed state of the controller
```
Comment (lpiwowar) [inline config/crd/bases/test.openstack.org_tobikoes.yaml:332]: ditto

**Review by lpiwowar — CHANGES_REQUESTED** [review]
Thank you @eshulman2 for the updates! I have just three small asks, please:

- an update of the `Fix lint` commit
- fix of the BackoffLimit comment
- fix the comment for the `CommonTestStatus` section

**Comment by lpiwowar** [comment]
@eshulman2, just FYI, I want to merge your PR first so that you do not have to face another conflicts -> https://github.com/openstack-k8s-operators/test-operator/pull/184#issuecomment-2358653738

File: controllers/tempest_controller.go [hunk 25]
```diff
@@ -343,11 +343,11 @@ func (r *TempestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 	//                 It will simplify the logic and duplicite code (Tempest vs Tobiko)
 	if externalWorkflowCounter < len(instance.Spec.Workflow) {
```
Comment (lpiwowar) [inline controllers/tempest_controller.go:344]: You want to add a similar piece of code into the AnsibleTest controller as well. Right now `NodeSelector` and `Tolerations` do not work in `workflow` section of `AnsibleTest CR`.

File: api/v1beta1/common.go [hunk 26]
```diff
@@ -16,48 +16,104 @@ limitations under the License.
...
+	// +kubebuilder:default:=""
+	// A URL of a container image that should be used by the test-operator for tests execution.
+	ContainerImage string `json:"containerImage"`
 
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// +kubebuilder:validation:optional
-	// +kubebuilder:default:=""
-	// Config map subpath for mounting, defaults to configmap root.
-	SubPath string `json:"subPath"`
+	// BackoffLimimt allows to define the maximum number of retried executions (defaults to 0).
```
Comment (lpiwowar) [inline api/v1beta1/common.go:69]:
```suggestion
	// BackoffLimit allows to define the maximum number of retried executions (defaults to 0).
```

File: api/v1beta1/ansibletest_types.go [hunk 27]
```diff
@@ -89,51 +79,22 @@ type AnsibleTestSpec struct {
...
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// BackoffLimit allows to define the maximum number of retried executions (defaults to 0).
-	// +kubebuilder:validation:Optional
-	// +kubebuilder:default:=0
-	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
-	BackoffLimit *int32 `json:"backoffLimit"`
-
-	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// A parameter that contains a workflow definition.
+	// A parameter  that contains a workflow definition.
```
Comment (lpiwowar) [inline api/v1beta1/ansibletest_types.go:89]:
```suggestion
	// A parameter that contains a workflow definition.
```

File: api/v1beta1/common.go [hunk 28]
```diff
@@ -117,16 +117,3 @@ type CommonTestStatus struct {
...
-	// +kubebuilder:validation:optional
-	// +optional
-	// Use with caution! This parameter specifies whether test-operator should spawn test
-	// pods with allowedPrivilegedEscalation: true and the default capabilities on
-	// top of capabilities that are usually needed by the test pods (NET_ADMIN, NET_RAW).
-	// This parameter is deemed insecure but it is needed for certain test-operator
-	// functionalities to work properly (e.g.: extraRPMs in Tempest CR, or certain set
-	// of tobiko tests).
-	Privileged *bool `json:"privileged,omitempty"`
-}
```
Comment (lpiwowar) [inline api/v1beta1/common.go:44]: This should not be part of the `Fix linting` commit (this probably belongs into the `Embed common struct` commit.)

**Review by lpiwowar — CHANGES_REQUESTED** [review]
@eshulman2, I'm sorry there are still some things that need to be addressed. You can ping me if you have a question about some of the comments.

File: controllers/tempest_controller.go (hunk 25 above)
Comment (eshulman2) [reply controllers/tempest_controller.go:344]: this is not actually needed for the ansibleTest as far as I see it but I can add it for uniformity.

File: api/v1beta1/common.go (hunk 28 above)
Comment (eshulman2) [reply api/v1beta1/common.go]: moved to the proper commit

**Comment by eshulman2** [comment]
recheck

File: api/v1beta1/common.go [hunk 29]
```diff
@@ -16,48 +16,104 @@ limitations under the License.
...
-	// This parameter is deemed insecure but it is needed for certain test-operator
-	// functionalities to work properly (e.g.: extraRPMs in Tempest CR, or certain set
-	// of tobiko tests).
-	Privileged bool `json:"privileged"`
+	// +kubebuilder:default:=""
+	// Config map subpath for mounting, defaults to configmap root.
+	SubPath string `json:"subPath"`
 }
 
-type WorkflowCommonParameters struct {
```
[the remark this answers is not in the document]
Comment (lpiwowar) [reply api/v1beta1/common.go:33]: Yes, I know. The thing is that we need to have the `privileged:` parameter with type `*bool` in the `workflow:` section. The reason for this is that we need to be able to distinguish whether the user set `spec.workflow.privileged` or not.

The logic that decides whether to use to value from the workflow section is build on the idea of workflow parameters being pointers [1]. We have a PR in progress that will unify the whole processing of the `workflow` section:

- https://github.com/openstack-k8s-operators/test-operator/pull/168

Because of that I suggest let's add the `WorkflowCommonParameters` section and deal with the rest in the PR 168. This is PR is more than helpful and it would be great if we could merge it.

[1] https://github.com/openstack-k8s-operators/test-operator/blob/main/controllers/tobiko_controller.go#L310

[the remark this answers is not in the document]
Comment (eshulman2) [reply api/v1beta1/common.go:33]: ack I changed it for all workflow parameters.

File: api/v1beta1/common.go [hunk 30]
```diff
@@ -41,23 +143,39 @@ type WorkflowCommonParameters struct {
 	// functionalities to work properly (e.g.: extraRPMs in Tempest CR, or certain set
 	// of tobiko tests).
 	Privileged *bool `json:"privileged,omitempty"`
-}
 
-type extraConfigmapsMounts struct {
+		// +operator-sdk:csv:customresourcedefinitions:type=spec
```
Comment (lpiwowar) [inline api/v1beta1/common.go:147]: Indentation:
```suggestion
     // +operator-sdk:csv:customresourcedefinitions:type=spec
```

File: api/v1beta1/common.go [hunk 31]
```diff
@@ -16,7 +16,31 @@ limitations under the License.
 
 package v1beta1
 
-type CommonParameters struct {
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ExtraConfigmapsMounts struct {
```
[the remark this answers is not in the document]
Comment (lpiwowar) [reply api/v1beta1/common.go:24]: @fmount this PR is solely about reorganizing the parameters inside the test-operator related CRs. This is a good point! I just think it can be addressed in a separate PR. If @eshulman2 finds the time to address this issue I'm would be more then happy if this got resolved but I know that she's already spent a significant amount of time on making the jobs happy.

I'm still going to do a proper review after the recent changes but unless we find something significant then from my point of view this is good to be merged.

File: api/v1beta1/common.go [hunk 32]
```diff
@@ -41,23 +143,39 @@ type WorkflowCommonParameters struct {
...
+	StorageClass *string `json:"storageClass"`
 
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
-	// +kubebuilder:validation:Required
-	// Path within the container at which the volume should be mounted.
-	MountPath string `json:"mountPath"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=""
+	// A URL of a container image that should be used by the test-operator for tests execution.
+	ContainerImage *string `json:"containerImage"`
```
Comment (lpiwowar) [inline api/v1beta1/common.go:157]: We are encountering regression. `ContainerImage` is not working in workflow after this change. The issue lays probably here [1]. The type is no longer `string` but `*string`. 

```
---
apiVersion: test.openstack.org/v1beta1
kind: AnsibleTest
metadata:
  namespace: openstack
  name: ansible-tests
spec:
  ...
  containerImage: "quay.io/lpiwowar0/openstack-tempest-all:root"
  ...
  workflow:
    - stepName: "tortuga"
      containerImage: "quay.io/lpiwowar0/openstack-tempest-all:workflow"
      ...
```

[1] https://github.com/openstack-k8s-operators/test-operator/blob/458fbd9e1616c14c6093802f473cbc1c1ebf7367/controllers/ansibletest_controller.go#L352C12-L352C29

**Review by lpiwowar — CHANGES_REQUESTED** [review]
I went through all the parameters and checked whether the behavior is still the same. Unfortunately the `containerImage:` parameter in `AnsibleTest` CR now behaves differently. This should be fixed.

Also, I'm thinking it is time to finally introduce some kuttle tests because checking the parameters can be quite tedious work.

File: api/v1beta1/common.go (hunk 31 above)
[the remark this answers is not in the document]
Comment (eshulman2) [reply api/v1beta1/common.go:24]: as lukas mentioned this is a reorg, @fmount please raise a jira ticket for this comment to make sure it won't get lost, I'm going to resolve this thread for now

[the remark this answers is not in the document]
Comment (fmount) [reply api/v1beta1/common.go:24]: Done: https://issues.redhat.com/browse/OSPRH-10366 @lpiwowar I assigned it to you but feel free to revisit that jira and coordinate the requiered work.

[the remark this answers is not in the document]
Comment (lpiwowar) [reply api/v1beta1/common.go:24]: Thanks @fmount !

**Review by lpiwowar — APPROVED** [review]
Looks good to me! :+1: 

The day has finally come :balloon: :tada: . I think this is ready to be merged.
